File: FSharp/ListFiltering3.fs
//Create List from 1 to 100, leave numbers greter tha  and sort so that even number were in top of the list
let NumbersPow = [ for x in 1 .. 100->x]
let list = NumbersPow|>List.filter(fun x->x > 50)
let sort(number1:int)(number2:int) = 
    if(number1%2 = 0)
        then -1
    else if(number2%2 = 0)
        then 1
    else 
        0

[<EntryPoint>]
let main argv =
    let result = List.sortWith sort list
    printfn "%A" result
    0 // return an integer exit code

File: JS/ListFiltering3.js

(function() {
    var data = (function () {
        var arr = [];
        for (var i = 0; i < 100; i++) {
            arr[i] = i;
        }
        return arr;
    })().filter(function(x) {
        return x > 50;
    });
    
    var sort = function(a, b) {
        if (a % 2 === 0) {
            return -1;
        }
        else if (b % 2 === 0) {
            return 1;
        } else {
            return 0;
        }
    }

    var main = function (list, func) {
        return data.sort(func);
    }

    console.log(main(data, sort));
})();

while (true) {
    
}
